index/store/boltdb/writer.go

```diff
@@ -38,7 +38,8 @@ func (w *Writer) ExecuteBatch(batch store.KVBatch) error {
 	if err != nil {
 		return err
 	}
-
+	defer tx.Rollback()
```

Comment:
I am not super familiar with the code, but shouldn't this go in the if block above?